mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -2113,6 +2139,53 @@ impl ElectrumClient {
                 }),
         )
     }
+
+    pub(crate) fn get_servers_with_latest_block_count(&self) -> UtxoRpcFut<(Vec<String>, u64)> {
+        let ticker = self.coin_ticker.clone();
+        let selfi = self.clone();
+        let fut = async move {
+            let connections = selfi.connections.lock().await;
+            let futures = connections
+                .iter()
+                .map(|connection| {
+                    let addr = connection.addr.clone();
+                    selfi
+                        .get_block_count_from(&addr)
+                        .map(|response| (addr, response))
+                        .compat()
+                })
+                .collect::<Vec<_>>();
+            drop(connections);
+
+            let responses = join_all(futures).await;
+
+            // First, we use filter_map to get rid of any errors and collect the
+            // server addresses and block counts into two vectors
+            let (responding_servers, block_counts_from_all_servers): (Vec<_>, Vec<_>) =
+                responses.clone().into_iter().filter_map(|res| res.ok()).unzip();
+
+            // Next, we use max to find the maximum block count from all servers
+            if let Some(max_block_count) = block_counts_from_all_servers.clone().iter().max() {
+                // Then, we use filter and collect to get the servers that have the maximum block count
+                let servers_with_max_count: Vec<_> = responding_servers
+                    .into_iter()
+                    .zip(block_counts_from_all_servers)
+                    .filter(|(_, count)| count == max_block_count)
+                    .map(|(addr, _)| addr)
+                    .collect();
+
+                // Finally, we return a tuple of servers with max count and the max count
+                return Ok((servers_with_max_count, *max_block_count));
+            }
+
+            return Err(MmError::new(UtxoRpcError::Internal(format!(
+                "Couldn't get block count from any server for {}, responses: {:?}",
+                ticker, responses
```

Comment:
You can avoid cloning ticker in upper block and use `&selfi.coin_ticker` here

Reply:
Done